gengokit/template/NAME-service/NAME-cli-client/client_main.gotemplate

```diff
@@ -31,27 +31,47 @@ var (
 
 
 func main() {
+	var (
+		httpAddr       = flag.String("http.addr", "", "HTTP address of addsvc")
+		grpcAddr       = flag.String("grpc.addr", ":5040", "gRPC (HTTP) address of addsvc")
+	)
+
+
 	// The addcli presumes no service discovery system, and expects users to
 	// provide the direct address of an service. This presumption is reflected in
 	// the cli binary and the the client packages: the -transport.addr flags
 	// and various client constructors both expect host:port strings.
 
-	var (
-		httpAddr       = flag.String("http.addr", "", "HTTP address of addsvc")
-		grpcAddr       = flag.String("grpc.addr", ":5040", "gRPC (HTTP) address of addsvc")
-		method         = flag.String("method", "{{range $index, $i := .Service.Methods}}{{if $index}}{{else}}{{ToLower $i.Name}}{{end}}{{end}}",	"{{range $index, $i := .Service.Methods}}{{if $index}},{{end}}{{ToLower $i.Name}}{{end}}")
-	)
+	{{ range $meth, $args := .ClientArgs.MethArgs }}
+	fs{{$meth}} := flag.NewFlagSet("{{ToLower $meth}}", flag.ExitOnError)
+	{{end}}
 
 	var (
 		{{.ClientArgs.AllFlags}}
 	)
-	flag.Parse()
 
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage of %s:\n", os.Args[0])
+		flag.PrintDefaults()
+		fmt.Fprintf(os.Stderr, "Subcommands:\n")
+	{{range $meth, $args := .ClientArgs.MethArgs }}
```

Comment:
So that there is not weird spacing in the generated code between these fmt.Fprints's.

```
{{range thing -}}
    //Stuff
{{end -}
```

Also I personally like indenting the generated code blocks as if they were code. Indentation will be fixed by gofmt so it makes it easier to read the template. For me at least, @hasIan has agreed in the past.